src/test/java/org/broadinstitute/hellbender/utils/tsv/TableReaderUnitTest.java

```diff
@@ -659,17 +680,29 @@ protected void processCommentLine(final String comment, final long ln) {
         Assert.assertEquals(actual, expected);
     }
 
-    private File createTestInput(final String... lines) throws IOException {
-        final File testFile = createTempFile("test", ".tab");
-        final PrintWriter testWriter = new PrintWriter(new FileWriter(testFile));
+    private Path createTestInput(final String... lines) throws IOException {
+        final Path testFile = createTempPath("test", ".tab");
+        fillTestInput(testFile, lines);
+        return testFile;
+    }
+
+    private Path createTestInputInMemory(final java.nio.file.FileSystem ramfs, final String... lines) throws IOException {
```

Comment:
Rather than adding a new "create" method with a `FileSystem` arg, add `createTestInputOnPath` with a `Path` arg, and have `createTestInput` pass a `Path` to that method. Then the jimfs test above can create the jimfs file directly and pass the pathto `createTestInputOnPath`. All the jimfs code will be confined to that one test, and fillTestInput can go away:
```suggestion
    private Path createTestInput(final String... lines) throws IOException {
        final Path testFile = createTempPath("test", ".tab");
        return createTestInputOnPath(testFile, lines);
    }

    private Path createTestInputOnPath(Path testPath, final String... lines) throws IOException {
        try (final PrintWriter testWriter = new PrintWriter(Files.newBufferedWriter(testPath));) {
            for (final String line : lines) {
                testWriter.println(line);
            }
        }
        return testPath;
    }
```

Reply:
OK, done! I didn't merge your suggestion (because that would undo the try-with-resources I just put in), but I copied it by hand and you should see it match in spirit.